pkg/server/storage_api/storage_api_test: TestDecommissionSelf failed

pkg/server/storage_api/storage_api_test.TestDecommissionSelf [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11586617?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11586617?buildTab=artifacts#/) on master @ [e0bdc9905d5275d5f12e7f289e2f00600e845096](https://github.com/cockroachdb/cockroach/commits/e0bdc9905d5275d5f12e7f289e2f00600e845096):


```
=== RUN   TestDecommissionSelf
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/800529510c8a77f305555fe4848c57f0/logTestDecommissionSelf2841017780
    test_log_scope.go:81: use -show-logs to present logs inline
    decommission_test.go:705: 
        	Error Trace:	pkg/server/storage_api/storage_api_test_test/pkg/server/storage_api/decommission_test.go:705
        	Error:      	Received unexpected error:
        	            	rpc error: code = Internal desc = failed to connect to n1 at 127.0.0.1:40673: initial connection heartbeat failed: grpc: n5 was permanently removed from the cluster at 2023-09-01 13:10:20.949947889 +0000 UTC; it is not allowed to rejoin the cluster [code 7/PermissionDenied], node is decommissioned
        	Test:       	TestDecommissionSelf
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/800529510c8a77f305555fe4848c57f0/logTestDecommissionSelf2841017780
--- FAIL: TestDecommissionSelf (11.83s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/unowned
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDecommissionSelf.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-31135

pkg/server/storage_api/storage_api_test.TestDecommissionSelf [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11869226?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/11869226?buildTab=artifacts#/) on master @ [a3dccc6eeb0fb00634806ded691f74f1835694f0](https://github.com/cockroachdb/cockroach/commits/a3dccc6eeb0fb00634806ded691f74f1835694f0):


```
=== RUN   TestDecommissionSelf
    test_log_scope.go:167: test logs captured to: /artifacts/tmp/_tmp/800529510c8a77f305555fe4848c57f0/logTestDecommissionSelf1229217120
    test_log_scope.go:81: use -show-logs to present logs inline
    decommission_test.go:705: 
        	Error Trace:	pkg/server/storage_api/storage_api_test_test/pkg/server/storage_api/decommission_test.go:705
        	Error:      	Received unexpected error:
        	            	rpc error: code = Internal desc = failed to connect to n1 at 127.0.0.1:39953: initial connection heartbeat failed: grpc: n5 was permanently removed from the cluster at 2023-09-21 12:51:21.21636237 +0000 UTC; it is not allowed to rejoin the cluster [code 7/PermissionDenied], node is decommissioned
        	Test:       	TestDecommissionSelf
    panic.go:522: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/800529510c8a77f305555fe4848c57f0/logTestDecommissionSelf1229217120
--- FAIL: TestDecommissionSelf (12.31s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
, <code>stress=true</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestDecommissionSelf.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/server/storage_api/decommission_test.go
+++ b/pkg/server/storage_api/decommission_test.go
@@ -657,6 +657,7 @@ func TestDecommissionSelf(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 	skip.UnderRace(t) // can't handle 7-node clusters
+	skip.UnderDeadlockWithIssue(t, 112918)
 
 	// Set up test cluster.
 	ctx := context.Background()

